protocol/x/clob/types/match_with_orders.go

```diff
@@ -73,8 +73,9 @@ func (match MatchWithOrders) Validate() error {
 	}
 
 	// Make sure the maker order is not an IOC order.
-	if makerOrder.MustGetOrder().TimeInForce == Order_TIME_IN_FORCE_IOC {
-		return errors.New("IOC order cannot be matched as a maker order")
+	unwrappedMakerOrder := makerOrder.MustGetOrder()
+	if unwrappedMakerOrder.RequiresImmediateExecution() {
+		return errors.New("IOC / FOK order cannot be matched as a maker order")
```

Comment:
I can update these errors to include the invalid order if anyone would prefer that.

Reply:
Including the order in the error sounds good to me